src/features/auth/sign-up/screen/index.tsx

```diff
@@ -1,13 +1,113 @@
-import React from 'react';
-import { Text, View } from 'react-native';
+import React, { useState } from 'react';
+import { Image, ImageBackground, Text, TextInput, TouchableOpacity, View } from 'react-native';
 
+import { AuthStackScreens } from 'navigation/stacks/auth';
+
+import { SignUpRequest } from 'network/models/user-models';
+import { useSignUp } from 'network/queries/user-queries';
+
+import common from '../../../../localization/resources/en.json';
 import styles from './styles';
 import { SignUpNavigationProps } from './types';
 
-const SignUpScreen: React.FunctionComponent<SignUpNavigationProps> = () => (
-  <View style={styles.container}>
-    <Text>Sign Up</Text>
-  </View>
-);
+const SignUpScreen: React.FunctionComponent<SignUpNavigationProps> = props => {
+  const [email, setEmail] = useState('');
+  const [name, setName] = useState('');
+  const [password, setPassword] = useState('');
+  const [passwordConfirmation, setPasswordConfirmation] = useState('');
+  const [error, setError] = useState('');
+
+  const { mutate } = useSignUp({
+    onError: () => {
+      setError(common.screen.signUp.errorAPI);
+    },
+    onSuccess: () => {
+      props.navigation.navigate(AuthStackScreens.Welcome);
+    },
+  });
+
+  const onSignUpPress = () => {
+    if (!email || !name || !password || !passwordConfirmation) {
+      setError(common.screen.signUp.errorEmpty);
+      return;
+    }
+    if (email.includes('@') === false) {
+      setError(common.screen.signUp.errorEmail);
+      return;
+    }
```

Comment:
hehe. just fyi => https://www.freecodecamp.org/news/react-form-validation-zod-react-hook-form/
React Hook Form + Zod validators play very well here, no change required tho

Reply:
Nice, I'll look into it. Thanks